src/app/PDF.tsx

```diff
@@ -254,22 +224,61 @@ interface PDFUIProps {
   children?: React.ReactNode;
 }
 
+export interface RenderAnnotationsHandler {
+  onMouseUp: React.MouseEventHandler;
+  // What should we do when we click down on the Canvas?
+  onMouseDown: React.MouseEventHandler;
+  // What should we do when our mouse moves on the canvas?
+  onMouseMove: React.MouseEventHandler;
+}
+
 // Actually render the PDFUI onto the screen. Ideally, the code in this component
 // should be extremely simple; most of the complicated functionality gets pushed
 // to hooks in other places.
 const PDFUI: React.FC<PDFUIProps> = (props) => {
-  const { url, width, height, children } = props;
+  const { url, width, height } = props;
   const { canvas, loading } = useFetchPDFUI(url);
+  const [annotations, step, tool, tokens] = useSelector((state) => [
+    Object.values(state.annotations),
+    state.step,
+    state.tool,
+    state.tokens,
+  ]);
 
   const {
     cursor,
     container,
-    creationBounds,
+    creationState,
     onMouseLeave,
     onClick,
     ...handlers
   } = useHandlers();
 
+  let annotationsAndTokens = <></>;
```

Comment:
Could we refactor this component into several components? At the top, we can have `PDFUI`, but we'd replace the body with something that looks like:

```
<PDFUI>
    {/* logic ... */}
   <LayerGatekeeper />
     {/* more logic ... */}
</PDFUI>
```

Then, in `LayerGatekeeper`, we would have code that looks like:

```
const LayerGatekeeper = () => {
   switch (step) {
      case 0: {
          return <FieldLayerAnnotations />
      }
     case 1: {
         return <LabelLayerAnnotations />
     }
   }
}
```

Then, as we need to add more code for the layers, we can extend the switch statement as necessary. The reason I suggest this refactoring is that it concentrates the logic of deciding what we show when into a single place, decoupled from the other concerns `PDFUI` has. 

Does this idea make sense to you? I'm very open to feedback; I've used this pattern in my own projects and found it helpful, but we can absolutely go a different route if you prefer.

Reply:
Amazing suggestion. I fixed this :)